### ISSUE TITLE
NodeJS - loadFromDatalessJSON failing to load multiple images

### ISSUE THREAD

**Issue by Tsunami2069** [fabricjs/fabric.js#5644]
Hello.

I'm having an issue with the NodeJS side of my project.
Version is the most recent one (2.7.0).

I have to load multiple canvas, each possibly containing multiple images, using loadFromDatalessJSON.
However, what has been happening is that when a canvas has multiple images of the same file type, he only seems to successfuly load the last one. It's as if each loading is breaking the previous one.

I'm attaching a screenshot of an example of this occuring. In this example, I'm executing 4 canvas loads. I've drawn some white lines separating each canvas load.

As you can see, the first canvas loads both images successfuly (but i think it only happens cause they're of different file types, one is JPG and the other is PNG).
All other canvas loads try to load 2 or 3 images but they fail the first 1 or 2 and only successfuly load the last one.

What could be causing this?

Also, just a little side question. I see that the most current version is 2.7.0 but the fabricjs.com website and the documentation still say 2.4.6.
Any particular reason for this?
Is 2.7.0 not officially out yet? Or maybe the site just hasn't been updated yet?
If 2.7.0 is officially out, where can I find the change log?
Just wondering cause I wanna make sure I use the most recent stable version.
Right now i'm using 2.4.5 on the browser side and 2.7.0 on the nodejs side.

Thank you for your help.

![Snip1](https://user-images.githubusercontent.com/14236716/56383106-eefbf800-6210-11e9-8af1-c96403cb8629.jpg)

**Comment by asturur** [comment]
i think you should try with the latest master, nodecanvas 2 works better.

Also would be nice to have a reproducible snippet.

**Comment by Tsunami2069** [comment]
@asturur 

I'm sorry, pardon my ignorance but could you give me just a bit more info / help on how I can try what you suggested?
I'm not the most savy person when it comes to NodeJS.

**Comment by asturur** [comment]
I'm sorry i have been out for long.
You can install the latest master cloning the repository from git and building the lib on your own.

you can specify the fabricJS dependency as:

"git+https://git@github.com/fabricjs/fabric.js.git"

or using `npm install fabric@git+https://git@github.com/fabricjs/fabric.js.git`

This as a workaround untill 3.0 is released

**Comment by Tsunami2069** [comment]
@asturur 

Thanks a lot for your help anyways.
I closed the issue cause I found a temporary solution to the problem.
Was gonna post something about it but then something came up and I ended up forgetting.

After fiddling here and trying there, I found out that if I replaced the full object.src inside the fromObject function, which comes with a full URL (including http://)  to a relative URL it would no longer cause the loading errors.
Maybe a cross-origin issue?

Anyways, doing the replacement fixed my problem.
